### PR TITLE
#134249123 Environment setup for frontend testing

### PR DESCRIPTION
**Pull request by andela-emurphy**
#### What does this PR do?   
  Setup unit testing for the frontend
#### Description of Task to be completed?
  Sets up the environment needed for testing frontend codes using jasmine and karma 
#### How should this be manually tested?
   Run "karma start"

### CODE REVIEW

File: jasmine/spec/test.spec.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+function sqrt(x) {
+  return x * x
+}
+
+describe('simple test ' , function() {
+
```
Comment (andela-kakpobome) [inline jasmine/spec/test.spec.js]: ### Nitpick

Unnecessary space here.

File: jasmine/spec/test.spec.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  it('should return sqrt ', function() {
+    var root = sqrt(4)
+    expect(root).toEqual(16)
+  })
+
```
Comment (andela-kakpobome) [inline jasmine/spec/test.spec.js]: ### Nitpick

Unnecessary space here.

File: .env [hunk 3]
```diff
@@ -0,0 +1,9 @@
+MONGOTEST_URL=TEST_DB
```
Comment (andela-kakpobome) [inline .env]: ### Critical

This file should not be committed to version control

File: server.js [hunk 4]
```diff
@@ -56,7 +58,7 @@ require('./config/routes')(app, passport, auth);
 
 //Start the app by listening on <port>
 var port = config.port;
-var server = app.listen(port);
+var server = app.listen(9000);
```
Comment (andela-kakpobome) [inline server.js]: ### Critical

This port number should be referencing it's value from the config file.

File: server.js [hunk 5]
```diff
@@ -58,7 +58,7 @@ require('./config/routes')(app, passport, auth);
 
 //Start the app by listening on <port>
 var port = config.port;
```
Comment (andela-kakpobome) [inline server.js]: ### Nitpick

This line is unnecessary, `config.port` can be used directly.

File: config/env/all.js [hunk 6]
```diff
@@ -1,12 +1,10 @@
 'use strict';
-
-var path = require('path'),
-rootPath = path.normalize(__dirname + '/../..');
-var keys = rootPath + '/keys.txt';
-var testLocal = 'mongodb://localhost/cfh';
+const path = require('path'),
+      rootPath = path.normalize(__dirname + '/../..'),
+      keys = rootPath + '/keys.txt';
```
Comment (adebayo-maborukoje) [inline config/env/all.js]: is this `keys` being used anywhere in this file?

File: config/env/all.js [hunk 7]
```diff
@@ -1,12 +1,10 @@
...
 
 module.exports = {
   root: rootPath,
   port: process.env.PORT || 3000,
-  db: process.env.MONGOHQ_URL || testLocal
+  db: process.env.MONGOHQ_URL || "mongodb://localhost/cfh"
```
Comment (adebayo-maborukoje) [inline config/env/all.js]: use single quote for Js files and double quotes in html files

File: server.js [hunk 8]
```diff
@@ -14,20 +17,20 @@ var express = require('express'),
 
 //Load configurations
 //if test env, load example file
-var env = process.env.NODE_ENV = process.env.NODE_ENV || 'development',
+let env = process.env.NODE_ENV = process.env.NODE_ENV || 'development',
```
Comment (adebayo-maborukoje) [inline server.js]: Do you intend to do something with the `env` here? i don't see it being used anywhere.
Also, the comment preceding that line suggests there is a condition statement coming. but i don't see it. Let your comments only reflects what is on happening.

File: test/server/article/model.js [hunk 9]
```diff
@@ -1,15 +1,16 @@
+'use strict';
 /**
  * Module dependencies.
  */
-var should = require('should'),
-    app = require('../../server'),
+const should = require('should'),
+    app = require('../../../server'),
```
Comment (adebayo-maborukoje) [inline test/server/article/model.js]: another unused `app` here. Kindly check all comments by `hound` when modifying your PRs

File: .env (hunk 3 above)
Comment (andela-emurphy) [inline .env]: it has been removed

File: server.js (hunk 4 above)
Comment (andela-emurphy) [inline server.js]: changed the port to the actual value

File: test/client/test.spec.js [hunk 10]
```diff
@@ -0,0 +1,12 @@
+'use strict';
+
+function sqrt(x) {
```
Comment (andela-emurphy) [inline test/client/test.spec.js]: corrected the name

File: config/env/all.js (hunk 6 above)
Comment (andela-emurphy) [inline config/env/all.js]: none that I know of. I removed it

File: jasmine/spec/test.spec.js (hunk 1 above)
Comment (andela-emurphy) [inline jasmine/spec/test.spec.js]: fixed

File: jasmine/spec/test.spec.js (hunk 2 above)
Comment (andela-emurphy) [inline jasmine/spec/test.spec.js]: fixed

File: test/server/article/model.js (hunk 9 above)
Comment (andela-emurphy) [inline test/server/article/model.js]: removing that line caused the test fail

**Comment by andela-kakpobome** [comment]
@andela-emurphy your PR messages should be able to answer the question `This commit should:`, please refactor the title of this one

File: server.js (hunk 5 above)
Comment (andela-emurphy) [inline server.js]: fixed

Comment (andela-emurphy) [inline server.js]: fixed

**Comment by andela-kakpobome** [comment]
@adebayo-maborukoje this PR title was supposed to be refactored before it was merged, to answer the question `This commit should`

**Comment by adebayo-maborukoje** [comment]
@andela-kakpobome My bad. I should have gone through all the comments again.

**Pull request merged**